paltax/source_models.py

```diff
@@ -293,16 +294,20 @@ def modify_cosmology_params(
         Returns:
             Modified cosmology parameters.
         """
-        npz_file = np.load(self.cosmos_path)
-        images = npz_file['images']
+
+        hdf5_file = h5py.File(self.cosmos_path, 'r')
```

Comment:
Since we're switching to hdf5, I propose three changes.

1. The cosmos catalog should now include a parameter that determines how many images to load when modify_cosmology_params is called. It should have a relatively large default value (say 2048).
2. The hdf5_file should be opened in init and saved as a self property. You should use [atexit](https://docs.python.org/3/library/atexit.html#atexit.register) at the end of the __init__ function to register a cleanup function that will close the hdf5 file once python exits. 
3. You should save a tracker of the number of times modify_cosmology_params has been called and the total length of the hdf5 file. The idea here is that each time we call modify_cosmology_params we should be accessing a new chunk of the hdf5 file. When we run out of chunks, we should start again at the first chunk. Happy to talk on slack if you need more details on this.

The high-level idea here is to make sure that we can use hdf5 files that are larger than the number of images we can realistically store on memory. Note that once you've made that change, we'll also have to change how you currently deal with the weights.